### ISSUE TITLE
sorting does not work when combined with other filters

### ISSUE THREAD

**Issue by stefanprobst** [acdh-oeaw/apis-core#339]
(i) the `sort` query parameter is not documented in openapi, but i correctly guessed that the following will sort persons by their name:

```
curl "https://ica.acdh-dev.oeaw.ac.at/apis/api/entities/person/?sort=name" | jq '.results[].name'

"Aaron"
"Adamic"
"Anderson"
"Apel"
"Armstrong"
"Berger"
"Bishop"
"Borden"
"Brand"
"Brown"
"Byrnes"
"Caldwell"
```

(ii) however, when adding an additional query parameter, search results are no longer sorted:

```
curl "https://ica.acdh-dev.oeaw.ac.at/apis/api/entities/person/?sort=name&collection=13" | jq '.results[].name'

"Beer-Hofmann"
"Aaron"
"Barsis"
"Aaron"
"Sachs-Barsis"
"Beer-Hofmann Lens"
"Benatzky"
"Benedikt"
"Boyle"
"Burlingham Tiffany"
"Burnett"
"Doolittle"
```

(iii) confusingly, the second request seems to include names which are not returned in the first request, e.g. "Beer-Hofmann" - not sure what that means?

**Comment by stefanprobst** [comment]
@sennierer any insights on this?

**Comment by richardhadden** [comment]
Looking into this now: it seems at first glance that `sort` doesn't work at all.

Using MPR data, the first 55 results are sorted, then the rest are in some random order:

```
"Accerboni, Joseph Peter"
...
"Arnold, Johann"
"Laurentini"
```

I guess this is just the order they are stored in the database — with the result that any filters end up showing results that go beyond the top few which happen to be in the right order. 

@stefanprobst Can you confirm this in some way?

**Comment by stefanprobst** [comment]
> I guess this is just the order they are stored in the database

yep, i guess that is what's happening. ICA backerd also seems to go unsorted at index 53, see:

```
curl "https://ica.acdh-dev.oeaw.ac.at/apis/api/entities/person/?sort=name&limit=200" | jq '.results[].name ' | cat -n
```

**Comment by richardhadden** [comment]
I think this fix should resolve the problem. It seems the drf `OrderingFilter` had not been added to `filter_backends` (`apis_core/api_routers.py` line 453). I have added this now and it seems to work (along with other query filters).

The drf param for sorting is `ordering=`. Change this in project settings using `REST_FRAMEWORK["ORDERING_PARAM"]`

**Comment by steffres** [comment]
Richard's PR is merged into main and so `ordering=` can be used now.

@stefanprobst please check and close this issue if verified.

**Comment by stefanprobst** [comment]
the above query:

```bash
curl "https://ica.acdh-dev.oeaw.ac.at/apis/api/entities/person/?sort=name&limit=200" | jq '.results[].name ' | cat -n
```

is now sorted, thx.

how to sort in descending order?

**Comment by richardhadden** [comment]
> how to sort in descending order?

put a - before the parameter

(see https://www.django-rest-framework.org/api-guide/filtering/)

**Comment by stefanprobst** [comment]
actually no - there is still something wrong. please try:

```bash
curl "https://ica.acdh-dev.oeaw.ac.at/apis/api/entities/person/?ordering=-name&limit=500" | jq '.results[].name ' | cat -n
```

**Comment by stefanprobst** [comment]
are apis instances updated with this fix?

**Comment by stefanprobst** [comment]
can you post an example request where this is working?

**Comment by richardhadden** [comment]
> can you post an example request where this is working?

Not sure. I think @steffres merged this change, but maybe only into `dev`, so it may not be deployed?

**Comment by steffres** [comment]
I merged it, but am not aware of the current state of CI/CD and everything autodeployment-wise and github migration. 

The PR is both in `dev` and `main`, so probably just not deployed?

**Comment by stefanprobst** [comment]
if auto-deployment is not working - are you fixing it? should i open a new issue? where would i check if something failed?

**Comment by steffres** [comment]
@sennierer what is the status on CI/CD on github please? 

In this issue here, a PR provides fix but apparentely it's not yet deployed.
